src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -47,23 +54,42 @@ public String send(Letter letter, String serviceName) throws JsonProcessingExcep
 
         log.info("Generated message: id = {} for letter with print queue id = {} ", messageId, letter.type);
 
-        sendClient
-            .sendAsync(createQueueMessage(new WithServiceName<>(letter, serviceName), messageId))
-            .thenRunAsync(() -> {
-                log.info("Message acknowledged: id = {}", messageId);
-            })
-            .thenRunAsync(sendClient::closeAsync);
+        Message message = createQueueMessage(new WithServiceName<>(letter, serviceName), messageId);
+        Instant startSending = Instant.now();
+
+        CompletableFuture<Void> sendResult = sendClient
+            .sendAsync(message)
+            .whenCompleteAsync((result, exc) -> {
+                Duration tookSending = Duration.between(startSending, Instant.now());
+                insights.trackMessageAcknowledgement(tookSending, exc != null, messageId);
+
+                if (exc != null) {
+                    insights.trackException((Exception) exc);
+                } else {
+                    log.info("Message acknowledged: id = {}", messageId);
+                }
+            });
+
+        sendResult.thenRunAsync(sendClient::closeAsync);
+
+        if (sendResult.isCompletedExceptionally()) {
+            throw new SendMessageException("Could not send message to ServiceBus with " + messageId);
+        }
 
         return messageId;
     }
 
     private Message createQueueMessage(WithServiceName<Letter> letter, String messageId)
         throws JsonProcessingException {
 
+        insights.trackMessageReceived(letter.service, letter.obj.template, messageId);
```

Comment:
I think message is received earlier than this point.At this point we are just converting it to a form which Azure API can understand.

Reply:
this is logging the incoming request to print some sort of letter. since serialisation can throw exception and we might loose the information of the message we want to track - the insights is done right before that. especially when it is not needed to track any time related metric